elements/pf-dropdown/pf-dropdown.ts

```diff
@@ -0,0 +1,216 @@
+  static readonly styles = [styles];
+  private static instances = new Set<PfDropdown>();
+
+  static {
+    document.addEventListener('click', function(event) {
+      for (const instance of PfDropdown.instances) {
+        if (!instance.noOutsideClick) {
+          instance.#outsideClick(event);
+        }
+      }
+    });
+  }
+
+  /**
+   * Don't hide the dropdown when clicking ouside of it.
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'no-outside-click' }) noOutsideClick = false;
+
+  /**
+   * Disable the dropdown trigger element
+   */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  #triggerElement: HTMLElement | null = null;
+
+  @query('slot[name="trigger"]') private triggerSlot!: HTMLSlotElement;
+  @query('pf-dropdown-menu') private menuElement!: HTMLElement;
+
+  #float = new FloatingDOMController(this, {
+    content: (): HTMLElement | undefined | null =>
+      this.shadowRoot?.querySelector('pf-dropdown-menu'),
+  });
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.#triggerElement?.setAttribute('aria-haspopup', 'true');
+    this.#triggerElement?.setAttribute('aria-controls', 'dropdown-menu');
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    PfDropdown.instances.delete(this);
+  }
+
+  render() {
+    return html`
+      <slot
+        part="dropdown-trigger"
+        ?disabled="${this.disabled}"
+        name="trigger"
+        id="trigger"
+        @keydown=${this.handleDropdownButton}
+        @click=${this.toggleMenu}
+      >
+        <pf-button ?disabled="${this.disabled}" variant="control">
+          Dropdown 
+          <svg viewBox="0 0 320 512" fill="currentColor" aria-hidden="true" width="1em" height="1em">
+            <path d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"></path>
+          </svg>
+        </pf-button>
+      </slot>
+      <pf-dropdown-menu
+        part="dropdown-menu"
+        class="${this.#float.open && !this.disabled ? 'show' : ''}"
+        @keydown=${this.onKeydown}
+        @click="${this.handleSelect}"
+        id="dropdown-menu"
+      >
+        <slot></slot>
+      </dpf-dropdown-menuiv>
+    `;
+  }
+
+  /**
+   * sets focus on trigger element
+   */
+  focus() {
+    this.menuElement?.focus();
+  }
+
+  #outsideClick(event: MouseEvent) {
+    const path = event?.composedPath();
+    if (!path?.includes(this)) {
+      this.hide();
+    }
+  }
+
+  /**
+   * Toggle the dropdown
+   */
+  @bound async toggleMenu() {
+    this.#float.open ? this.hide() : this.show();
+  }
+
+  #setTriggerElement() {
+    const hasAssignedNodes = this.triggerSlot?.assignedNodes()?.length > 0;
+    if (!hasAssignedNodes) {
+      const defaultButton = this.triggerSlot?.children[0] as HTMLElement;
+      this.#triggerElement = defaultButton;
+    } else {
+      this.#triggerElement = document.activeElement as HTMLElement;
+    }
+  }
```

Comment:
Maybe use `@queryAssignedElements` instead of `@query` for `this.triggerSlot` 

```js
import { queryAssignedElements } from 'lit/decorators/query-assigned-elements.js';
... 
@queryAssignedElements({ slot: 'trigger', flatten: true }) private _triggerElements?: HTMLElement[];

...

#setTriggerElement() {
  if (this._triggerElements) {
    const [defaultButton] = this. _triggerElements;
    this.#triggerElement = defaultButton;
  } else {
    this.#triggerElement = document.activeElement as HTMLElement;
  }
}
```